### PR TITLE
dulwich, urllib3 & hg-git: upgrade and fix dependencies

### PR DESCRIPTION
**Pull request by danchr**
#### Description

I wanted to use hg-git to fetch a port that had migrated to Git, but neglected to fix their build system. Unfortunately, that didn't work due to missing dependencies.

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 10.x
Xcode 8.x

###### Verification 
Have you

- [ ] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [ ] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [ ] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?

- [ ] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?

### CODE REVIEW

File: python/py-urllib3/Portfile [hunk 1]
```diff
@@ -26,6 +26,10 @@ checksums           rmd160  8e110cf646403b479129233ce7c16cff4f935576 \
 if {${name} ne ${subport}} {
     depends_build-append \
                         port:py${python.version}-setuptools
+    depends_lib-append \
+                        port:py${python.version}-brotli \
+                        port:py${python.version}-certifi \
+                        port:py${python.version}-ipaddress
```
Comment (stromnov) [inline python/py-urllib3/Portfile:32]: Specified dependencies are marked as 'extra' for urllib3 module:
```
extras_require={
        "brotli": ["brotlipy>=0.6.0"],
        "secure": [
            "pyOpenSSL>=0.14",
            "cryptography>=1.3.4",
            "idna>=2.0.0",
            "certifi",
            "ipaddress; python_version=='2.7'",
        ],
        "socks": ["PySocks>=1.5.6,<2.0,!=1.5.7"],
    },
)
```

(note: ipaddress is python27 only)

So I think it'll be better to add those dependencies directly to py-hggit port.

Comment (danchr) [reply python/py-urllib3/Portfile:32]: Good point, although `certifi` and the `secure` extra in general does seem reasonable to elevate, don't you think? (Accessing HTTPS URLs is pretty standard these days…)

**Comment by lbschenkel** [comment]
+1 to `py-dulwich` changes.

File: python/py-urllib3/Portfile (hunk 1 above)
Comment (ryandesign) [reply python/py-urllib3/Portfile:32]: Defaulting to adding https support seems like a good idea to me.

**Comment by ryandesign** [comment]
Where are we with finishing this PR? From the comments it sounds like some dependencies still need to be added or moved.

Some builds had failed because py38-brotli didn't exist, but it has been created in the mean time.

There's a conflict in py-urllib3 but only because it was updated to a new version.

**Comment by easye** [comment]
> Where are we with finishing this PR? From the comments it sounds like some dependencies still need to be added or moved.

MacPorts mercurial currently uses Python 2.7, so I think using any Python 3 libraries for mercurial extensions is not a good idea until we at least have a variant for Mercurial.

**Comment by ryandesign** [comment]
> MacPorts mercurial currently uses Python 2.7, so I think using any Python 3 libraries for mercurial extensions is not a good idea until we at least have a variant for Mercurial.

I don't really understand how your comment relates. This PR doesn't change which Python Mercurial uses. It just updates some ports and adds missing dependencies.

**Comment by easye** [comment]
> I don't really understand how your comment relates. This PR doesn't change which Python Mercurial uses. It just updates some ports and adds missing dependencies.

Ah, yes, I was wrong.  The ports that `hg-git` depends on can work under Python3 independently of what version of Python that mercurial uses.

File: python/py-urllib3/Portfile (hunk 1 above)
Comment (danchr) [reply python/py-urllib3/Portfile]: I moved the `brotli` dependency to `hg-git`.

**Comment by danchr** [comment]
> > I don't really understand how your comment relates. This PR doesn't change which Python Mercurial uses. It just updates some ports and adds missing dependencies.
> 
> Ah, yes, I was wrong. The ports that `hg-git` depends on can work under Python3 independently of what version of Python that mercurial uses.

I replaced it with a port that uses Python 3, and assumed maintainership  — is that ok with you?

File: python/py-hggit/Portfile [hunk 2]
```diff
@@ -1,44 +1,15 @@
 # -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
 
+# obsoleted 20200820
 PortSystem              1.0
-PortGroup               python 1.0
-PortGroup               bitbucket 1.0
+PortGroup               obsolete 1.0
 
-epoch                   20130201
+epoch                   20200820
```
Comment (ryandesign) [inline python/py-hggit/Portfile:7]: Don't increase the epoch. Do increase the revision.

**Comment by danchr** [comment]
Manually merged.

**Pull request closed without merging**